src/runner.js

```diff
@@ -28,12 +28,96 @@ const debugResponse = _debug('remote-s3:Runner:res');
  * actually transmitted matched a pre-computed size and hash
  */
 class Runner {
```

Comment:
Question why do you want to keep this a separate object from the client object...

Wouldn't it be fine to just modify `request` from `fast-azure-storage` to accept a buffer instead of a string and then use that?
And similarly steal the `retry` function and wrap with that...

https://github.com/taskcluster/fast-azure-storage/blob/master/lib/utils.js#L127